homeassistant/components/unifiprotect/media_source.py

```diff
@@ -107,7 +107,7 @@ def _get_month_start_end(start: datetime) -> tuple[datetime, datetime]:
 
 
 @callback
-def _bad_identifier(identifier: str, err: Exception | None = None) -> BrowseMediaSource:
+def _bad_identifier(identifier: str, err: Exception | None = None) -> NoReturn:
```

Comment:
`_bad_identifier_media` below seems affected.

Reply:
That one explicitly casts (which is an intentional override from the developer in general). I couldn't judge why, but I do agree that case is never happening... (unreachable code).

Hmm... looking at it a bit more, the original typing and the one below, seem to have done to satisfy typing in an incorrect way. Will adjust it.

Reply:
Did further cleanup in [`0f19659` (#85050)](https://github.com/home-assistant/core/pull/85050/commits/0f1965975a9cd64069b6ff11ca3b34fc8094bac6)